enhancements/update/available-update-metadata.md

```diff
@@ -0,0 +1,266 @@
+
+Then, in the cluster-version operator (CVO), this enhancement proposes porting existing logic like [available-update translation][cluster-version-operator-update-translation] and [available-update lookup][cluster-version-operator-update-lookup] to preserve the Cincinnati metadata, with information from the release image itself taking precedence over information from Cincinnati.
+In some cases (when an administrator requested an update that was not in the available update set), this would require an additional Cincinnati request to retrieve metadata about the requested release.
+
+#### Metadata properties
+
+This enhancement declares the following to be well-known `metadata` properties in Cincinnati node objects:
+
+* `url` (optional), a homepage for the release.
+    For releases created by Red Hat, this will usually be the Errata for the release.
+* `io.openshift.upgrades.graph.release.channels` (optional), the comma-delimited set of channels to which the release currently belongs.
+
+Cincinnati implementations may, at their discretion, define, add, and remove additional properties as they see fit.
+
+If Cincinnati removes those properties or assigns different semantics to those properties, then we will bump `apiVersion` on the cluster-side custom resource.
+That would be painful, so I'm glad there is nothing on the horizon that would lead to Cincinnati making changes like that.
+
+### User Stories
+
+#### Dynamic Channel Selection
+
+Alice's cluster is on release 4.1.23, and she wants to know if she can update to 4.2.
+With interfaces like the web console and `oc adm upgrade ...` exposing:
+
+```console
+$ curl -sH 'Accept:application/json' 'https://api.stage.openshift.com/api/upgrades_info/v1/graph?channel=stable-4.1&version=4.1.23' | jq -r '.nodes[] | select(.version == "4.1.23").metadata["io.openshift.upgrades.graph.release.channels"] | split(",")[]'
+prerelease-4.1
+stable-4.1
+candidate-4.2
+```
+
+she could see that, while upgrading to 4.2 was possible, it was not yet considered stable.
+And running the query again later, she could see that it had been added to `stable-4.2`.
+
+The web console and `oc` would use `status.desired.channels`, to provide this information.
+
+#### Available-Update Homepages
+
+Currently the web console only shows available update version names.
+With this enhancement, they could also provide *View release notes* links, to give users convenient access to errata or other homepages.
+
+```console
+$ curl -sH 'Accept:application/json' 'https://api.stage.openshift.com/api/upgrades_info/v1/graph?channel=stable-4.1&version=4.1.23' | jq -r '.nodes[] | select(.version == "4.1.23").metadata.url'
+https://access.redhat.com/errata/RHBA-2019:3766
+```
+
+The web console and `oc` would use `status.availableUpdates.homepage` to provide this information.
```

Comment:
Why is this named `homepage` when it's called `url` in URL?  Just not updated?

Call this `url` to match the field unless there is a good reason.

Reply:
> Why is this named `homepage` when it's called `url` in the graph?

URL/I-ness is already encoded in the property type.  I wanted a name that reflected the intended semantics, without just stuttering the type.

Reply:
```
// Homepage is the homepage for this release image.  For example,
 // this could link errata or other release notes describing the release
 // image.
  Homepage URL
```

Why does not Homepage have a clear definition i.e. only errata URL because Cincinnati currently only gives you errata URL? 

Also the name `Homepage` is ambiguous in this context. I would be ok with just `URL`.

Reply:
> Why does not Homepage have a clear definition i.e. only errata URL because Cincinnati currently only gives you errata URL?

Folks that don't get errata (does FCOS?) may want to link a non-errata homepage.

> Also the name Homepage is ambiguous in this context. I would be ok with just URL.

How is `URL URL` _less_ ambiguous than `Homepage URL`?

Reply:
Homepage is an introductory page of a website, typically serving as a table of contents for the site or web page set as the default or start-up page on a browser. (from google search)

Also after going through [1] [2] I am more convinced that we are not talking about the first landing page. In our case only users who wants to know more should visit the page. Which does not sound as homepage. For example errata URLs does not fall under homepage definations :)

[1] https://dictionary.cambridge.org/dictionary/english/home-page
[2] https://techterms.com/definition/home_page

Reply:
URL is a widely understood indication of the place you go to learn about something.

Reply:
"Best place for humans to learn more about this release" survey

What do we call this?
Caveat that for a number of these formats, this is for the per-project URI, not a per-release URI.

## URL

* [Maven][], with `<project><url>...</url>...</project>` for "The URL to the project's homepage".  Also defines `<organization><url>...</url>...</organization>`, etc. with other URI values with different content.
* [Open Containers image-spec][image-spec-annotations] has `url` for "URL to find more information on the image".  Also defines `documentation` and `source` with other URI values with different content.
* [RPM][].  Also defines `Source0` with a URI value with different content.

## Homepage

* [Homebrew formula][homebrew-formula].
* [Deb][].
* [ebuild][].
* [NPM][].
* [Python][] (as `Home-page`).
* [Rust][].

[Deb]: https://www.debian.org/doc/debian-policy/ch-controlfields.html#homepage
[ebuild]: https://projects.gentoo.org/pms/7/pms.html#x1-620007.3
[homebrew-formula]: https://docs.brew.sh/Formula-Cookbook#fill-in-the-homepage
[image-spec-annotations]: https://github.com/opencontainers/image-spec/blob/v1.0.1/annotations.md#pre-defined-annotation-keys
[Maven]: https://maven.apache.org/ref/3.6.3/maven-model/maven.html
[NPM]: https://docs.npmjs.com/files/package.json#homepage
[Python]: https://packaging.python.org/specifications/core-metadata/#home-page
[RPM]: https://rpm-packaging-guide.github.io/#preamble-items
[Rust]: https://doc.rust-lang.org/cargo/reference/manifest.html#the-homepage-field

Reply:
We fought this out out of band, and I've pushed eba7c52 -> 2eb16cf pivoting to `url`, with a section explain the choice under `alternatives`.